src/arch/fake_time/faketimerha.cpp

```diff
@@ -28,45 +35,80 @@ CFakeTimerHandler::~CFakeTimerHandler() {
 
 void CFakeTimerHandler::run() {
   while(isAlive()) {
-    if(sleepTime > 0) {
-      CEventChainExecutionThread *execThread = getExecThread(); //get execThread for every sleep fake, since resource potentially changed
-      while(sleepTime > 0) {
+    if(!sleepTimes.empty()) {
+      CFakeTimerHandler::napinfo nap = sleepTimes.front();
+      sleepTimes.pop();
+      if(nap.napDuration == 0 && nap.wakupTime < (getNanoSecondsMonotonic() / cNanosecondsToMilliseconds)) {
+        // We have to go back in time!
         nextTick();
-        trackFakeForteTime();
-        sleepTime--;
-        while(execThread != 0 && execThread->isProcessingEvents()) {
-          sleepThread(0);
+        jumpFakeForteTime(nap.wakupTime);
+        DEVLOG_DEBUG("[FAKETIME]: time-jumping to destination %d ms\n", nap.wakupTime);
+        sleepThread(0);
+      } else {
+        nap.napDuration = nap.wakupTime - getNanoSecondsMonotonic() / cNanosecondsToMilliseconds;
+        if(nap.napDuration > cHourInMilliseconds) {
+          // if a destination time is more than one hour away from the current time:
+          // skip to one hour before the destination and start simulation from there
+          // this is done to avoid timeouts when communicating with 4diac-ide
+          jumpFakeForteTime(nap.wakupTime - cHourInMilliseconds);
+          DEVLOG_DEBUG("[FAKETIME]: time-jumping to one hour before destination %d ms\n", nap.wakupTime - cHourInMilliseconds);
+          nap.napDuration = cHourInMilliseconds;
+        }
+        DEVLOG_DEBUG("[FAKETIME]: advance time for %d ms\n", nap.napDuration);
+        auto execThread = getExecThread(nap.fakeSleepFb);
+        while(nap.napDuration  > 0) {
+          nextTick();
+          trackFakeForteTime();
+          nap.napDuration--;
+          while(execThread != 0 && execThread->isProcessingEvents()) {
+            sleepThread(0);
+          }
+        }
+        if(nap.napDuration == 0) {
+          DEVLOG_DEBUG("[FAKETIME]: reached target time\n");
         }
       }
-      startOutputEvent();
+      startOutputEvent(nap.fakeSleepFb);
     } else {
-      sleepThread(1);
+      sleepThread(0);
     }
   }
 }
 
-CEventChainExecutionThread* CFakeTimerHandler::getExecThread() {
-    return fakeSleepFb ? fakeSleepFb->getResource()->getResourceEventExecution() : nullptr;
+void CFakeTimerHandler::sleepToTime(const CIEC_TIME &t, CEventChainExecutionThread *execThread) {
```

Comment:
execThread is not used anymore. Please remove.

Reply:
removed, solves the other point as well